pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -151,8 +158,14 @@ func New(chain Ledger, cfg config.NeoFSBlockFetcher, logger *zap.Logger, putBloc
 	if cfg.DownloaderWorkersCount <= 0 {
 		cfg.DownloaderWorkersCount = defaultDownloaderWorkersCount
 	}
-	if len(cfg.Addresses) == 0 {
-		return nil, errors.New("no addresses provided")
+	if cfg.IndexFileSize <= 0 {
```

Comment:
Second point from https://github.com/nspcc-dev/neo-go/issues/3718 is not done. `defaultTimeout` is 5 minutes, and timeout from config is 10 minutes. Check the rest of fields.

Reply:
Not sure about ` defaultOIDBatchSize = bqueue.DefaultCacheSize` as we do have `BQueueSize` as a parameter, but if it is not specified it will be the default and will work fine. But I do not want to change OIDBatchSize and BQueueSize and make them equal in the `config/protocol.*.yml` because the current version of it is more "right".

Reply:
A good note. BlockFetcher's queue seems to be kind of "special" queue from that it works with reliable source of blocks and the data flow is predictable. Hence, I consider we need to add `blockfetcher.DefaultQueueCacheSize` constant with a special meaning, and sanitize config before the queue creation. Then default value for `defaultOIDBatchSize` should be the half of configured BlockQueueSize, and we don't even need a designated constant for that.